pest/src/prec_climber.rs

```diff
@@ -83,7 +85,7 @@ impl<R: RuleType> BitOr for Operator<R> {
 /// [1]: https://en.wikipedia.org/wiki/Operator-precedence_parser#Precedence_climbing_method
 #[derive(Debug)]
 pub struct PrecClimber<R: RuleType> {
-    ops: HashMap<R, (u32, Assoc)>
+    ops: BTreeMap<R, (u32, Assoc)>
```

Comment:
I'm curious about this change. It seems like this isn't relevant for no_std. Why the change?

Reply:
It is relevant, HashMap isn't in `alloc` (it uses rand or something).

Reply:
Maybe it could be `cfg`'d though, if it were any slower (is it? benching required)

Reply:
I think that the best approach here is to limit the `Vec` in the contractor to 256. (assert + document with `# Panics` section) Then use a `Box<[Assoc; 256]>` to store the `ops`.

Reply:
We'll definitely want to test to see if there's a notable difference. I suspect for most of our use-cases it'll be similar enough.